crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -1245,27 +1241,94 @@ impl Segment2d {
     ///
     /// Panics if `point1 == point2`
     #[inline(always)]
+    #[deprecated(since = "0.16.0", note = "Use the `new` constructor instead")]
     pub fn from_points(point1: Vec2, point2: Vec2) -> (Self, Vec2) {
-        let diff = point2 - point1;
-        let length = diff.length();
+        (Self::new(point1, point2), (point1 + point2) / 2.)
+    }
 
-        (
-            // We are dividing by the length here, so the vector is normalized.
-            Self::new(Dir2::new_unchecked(diff / length), length),
-            (point1 + point2) / 2.,
-        )
+    /// Create a new `Segment2d` at the origin from a `direction` and `length`
+    #[inline(always)]
+    pub fn from_direction_and_length(direction: Dir2, length: f32) -> Segment2d {
+        let half_length = length / 2.;
+        Self::new(direction * -half_length, direction * half_length)
     }
 
     /// Get the position of the first point on the line segment
     #[inline(always)]
     pub fn point1(&self) -> Vec2 {
-        *self.direction * -self.half_length
+        self.vertices[0]
     }
 
     /// Get the position of the second point on the line segment
     #[inline(always)]
     pub fn point2(&self) -> Vec2 {
-        *self.direction * self.half_length
+        self.vertices[1]
+    }
+
+    /// Get the segment's center
+    #[inline(always)]
+    #[doc(alias = "midpoint")]
+    pub fn center(&self) -> Vec2 {
+        (self.point1() + self.point2()) / 2.
+    }
+
+    /// Get the segment's length
+    #[inline(always)]
+    pub fn length(&self) -> f32 {
+        self.point1().distance(self.point2())
+    }
+
+    /// Get the segment translated by the given vector
+    #[inline(always)]
+    pub fn translated(&self, translation: Vec2) -> Segment2d {
+        Self::new(self.point1() + translation, self.point2() + translation)
+    }
+
+    /// Compute a new segment, based on the original segment rotated around the origin
+    #[inline(always)]
+    pub fn rotated(&self, rotation: Rot2) -> Segment2d {
+        pub fn rotate_point(p: Vec2, rotation: Rot2) -> Vec2 {
+            Vec2::new(
+                p.x * rotation.cos - p.y * rotation.sin,
+                p.x * rotation.sin + p.y * rotation.cos,
+            )
+        }
+        Segment2d::new(
+            rotate_point(self.point1(), rotation),
+            rotate_point(self.point2(), rotation),
+        )
+    }
```

Comment:
This was meant to be just this
```suggestion
    /// Compute a new segment, based on the original segment rotated around the origin
    #[inline(always)]
    pub fn rotated(&self, rotation: Rot2) -> Segment2d {
        Segment2d::new(rotation * self.point1(), rotation * self.point2())
    }
```